### DISCUSSION TITLE
Proposal To Add Unit / Widget Tests

### DISCUSSION THREAD

**Discussion by saladthieves** [nank1ro/flutter-shadcn-ui#189] [Ideas]
First of all, thanks for the awesome library. I'm using it in a few desktop apps I'm working on and love it! 
I've been thinking of adding unit tests in my own apps for the widgets but realize it would be better if they were added to the `flutter-shadcn-ui` repo instead. Thoughts?

**Comment by nank1ro** [comment]
Yes, I completely agree! I haven't written them yet because the components are still at risk of change until I reach version 1.0.
But I think in general there are no big problems, if you or others want to start opening PRs by adding tests I would be super happy to review them.
For now, I personally prefer to concentrate on adding as many components as possible to bridge the gap with the official shadcn library.

To release the first stable release (v1.0) I had already planned that I would add tests and comments to all widgets.

**Reply by saladthieves** [reply]
> I haven't written them yet because the components are still at risk of change until I reach version 1.0.

Gotcha. After all, it would be counterproductive to add tests for components when the API hasn't yet stabilized. We could start by adding tests to core areas which other components depend on (or the areas where things are least likely to change) as well as add any documentation where necessary.

**Comment by nank1ro** [comment]
Some unit/widgets tests have been added. More should come.
